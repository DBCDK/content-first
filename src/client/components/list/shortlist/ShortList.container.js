import React from 'react';
import {connect} from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import {
  ON_SHORTLIST_REMOVE_ELEMENT,
  SHORTLIST_UPDATE_ORIGIN,
  SHORTLIST_CLEAR
} from '../../../redux/shortlist.reducer';
import {OPEN_MODAL} from '../../../redux/modal.reducer';
import {ORDER} from '../../../redux/order.reducer';
import BookCover from '../../general/BookCover.component';
import OrderButton from '../../order/OrderButton.component';
import Link from '../../general/Link.component';
import Toolbar from '../../base/Toolbar';
import Title from '../../base/Title';
import Text from '../../base/Text';
import T from '../../base/T';
import Divider from '../../base/Divider';
import Button from '../../base/Button';
import Banner from '../../base/Banner';
import {filterCollection} from '../../work/workFunctions';
import withWork from '../../base/Work/withWork.hoc';

export class ShortListItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.origin
    };
  }

  render() {
    const {work, pid, origin, className} = this.props;
    if (!work) {
      return null;
    }
    const url = `/værk/${pid}`;

    // get collections including ereolen
    const collection = filterCollection(this.props.work);
    const addToListButton = (
      <Button
        align="left"
        size="medium"
        type="tertiary"
        className="mr-2 text-uppercase"
        onClick={this.props.onClick}
      >
        <T component="list" name="addToList" />
      </Button>
    );
    const orderBookButton = (
      <OrderButton
        book={work.book}
        align="right"
        size="medium"
        type="quaternary"
        className="ml-2"
        iconLeft="chrome_reader_mode"
      >
        <T component="general" name="book" />
      </OrderButton>
    );
    const orderElectronicBookButtons =
      work.collectionHasLoaded &&
      collection.filter(col => col.count === 1).map(col => {
        return (
          <Button
            align="right"
            size="medium"
            type="quaternary"
            iconLeft={col.icon}
            className="ml-2"
            key={col.url}
            href={col.url}
          >
            {col.type}
          </Button>
        );
      });

    return (
      <React.Fragment>
        <div className={`short-list-item d-flex flex-row ${className}`}>
          <i
            className="material-icons remove-btn"
            onClick={this.props.onRemove}
          >
            clear
          </i>
          <Link href={url}>
            <BookCover
              book={work.book}
              style={{height: 'unset', width: '70px'}}
            />
          </Link>
          <div className="top-bar-dropdown-shortlist-item-page">
            <Title Tag="h1" type="title5" className="mr-4">
              {work.book.title}
            </Title>
            <Text type="body" className="mb-1">
              {work.book.creator}
            </Text>
            <Text type="body" className="mb-1">
              {origin}
            </Text>
            <Toolbar className="desktop-styling">
              {addToListButton}
              <Text align="right" type="body">
                <T component="work" name="loanTitle" />
              </Text>
              {orderBookButton}
              {orderElectronicBookButtons}
            </Toolbar>
            <div className="mobile-styling">{addToListButton}</div>
          </div>
        </div>
        <div className="mobile-styling">
          <Text align="left" type="body">
            <T component="work" name="loanTitle" />
          </Text>
          <Toolbar className="mobile-styling">
            <React.Fragment align="left">
              {orderBookButton}
              {orderElectronicBookButtons}
            </React.Fragment>
          </Toolbar>
        </div>
      </React.Fragment>
    );
  }
}

const ShortListItemWithWork = withWork(ShortListItem, {
  includeReviews: true,
  includeCollection: true,
  includeCover: true
});

class ShortList extends React.Component {
  render() {
    const {elements} = this.props.shortListState;
    return (
      <React.Fragment>
        <Banner
          color="#81c793"
          className="fixed-width-col-md position-relative text-uppercase"
          title={T({component: 'shortlist', name: 'title'})}
        />

        <div className="container">
          <div className="top-bar-dropdown-list-page col-11 col-centered">
            <div className="items mb-2">
              <Toolbar className="top-bar-upper-toolbar">
                <Button
                  align="right"
                  size="large"
                  type="quaternary"
                  iconLeft="delete"
                  className="bg-white pr-0"
                  onClick={() => this.props.clearList()}
                >
                  <T component="shortlist" name="shortlistClear" />
                </Button>
              </Toolbar>
              <Divider />
              <ReactCSSTransitionGroup
                transitionName="dropdownlist"
                transitionEnter={false}
                transitionLeaveTimeout={200}
              >
                {elements.map(e => (
                  <div key={e.book.pid}>
                    <ShortListItemWithWork
                      key={e.book.pid}
                      origin={e.origin}
                      pid={e.book.pid}
                      onRemove={() => {
                        this.props.remove(e.book.pid);
                      }}
                      onOriginUpdate={origin => {
                        this.props.originUpdate(origin, e.book.pid);
                      }}
                      onAddToList={() => {
                        this.props.addToList(
                          [
                            {
                              book: e.book,
                              description: e.origin || (
                                <T component="shortlist" name="origin" />
                              )
                            }
                          ],
                          this.props.isLoggedIn,
                          () => this.props.remove(e.book.pid)
                        );
                      }}
                    />
                    <Divider />
                  </div>
                ))}
              </ReactCSSTransitionGroup>
            </div>
            {elements.length === 0 && (
              <div className="empty-list-text">
                <T component="shortlist" name="emptyList" />
              </div>
            )}
            {elements.length > 0 && (
              <Toolbar className="bottom-toolbar mt-5 mb-5">
                <Button
                  align="right"
                  iconLeft="list"
                  size="large"
                  type="tertiary"
                  className="text-uppercase"
                  onClick={() =>
                    this.props.addToList(
                      elements,
                      this.props.isLoggedIn,
                      this.props.clearList
                    )
                  }
                >
                  <T component="list" name="addAllToList" />
                </Button>
                <Button
                  align="right"
                  iconLeft="chrome_reader_mode"
                  size="large"
                  type="quaternary"
                  className="btn ml-4"
                  onClick={
                    this.props.orderList.length > 0 &&
                    (() =>
                      this.props.orderAll(
                        this.props.orderList.map(e => e.book)
                      ))
                  }
                >
                  <T component="shortlist" name="shortlistOrder" />
                </Button>
              </Toolbar>
            )}
          </div>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => {
  const {elements} = state.shortListReducer;
  const orderList = (elements || [])
    .filter(o => {
      const order = state.orderReducer.orders[o.book.pid] || {};
      return order.orderState !== 'ordered';
    })
    .slice(0, 10);
  return {
    shortListState: state.shortListReducer,
    isLoggedIn: state.userReducer.isLoggedIn,
    orderList
  };
};

export const mapDispatchToProps = dispatch => ({
  orderAll: books => books.forEach(book => dispatch({type: ORDER, book})),
  remove: pid =>
    dispatch({
      type: ON_SHORTLIST_REMOVE_ELEMENT,
      pid
    }),
  originUpdate: (origin, pid) =>
    dispatch({
      type: SHORTLIST_UPDATE_ORIGIN,
      pid,
      origin
    }),
  addToList: (works, isLoggedIn, callback = null) =>
    dispatch({
      type: OPEN_MODAL,
      modal: isLoggedIn ? 'addToList' : 'login',
      context: isLoggedIn
        ? works
        : {
            title: 'Tilføj til liste',
            reason: 'Du skal logge ind for at flytte bøger til en liste.'
          },
      callback
    }),
  clearList: () =>
    dispatch({
      type: SHORTLIST_CLEAR
    })
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ShortList);
