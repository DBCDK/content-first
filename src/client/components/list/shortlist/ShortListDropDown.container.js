import React from 'react';
import {connect} from 'react-redux';
import TruncateMarkup from 'react-truncate-markup';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import {
  ON_SHORTLIST_EXPAND,
  ON_SHORTLIST_COLLAPSE,
  ON_SHORTLIST_REMOVE_ELEMENT
} from '../../../redux/shortlist.reducer';
import {ON_USERLISTS_COLLAPSE} from '../../../redux/list.reducer';
import {HISTORY_PUSH} from '../../../redux/middleware';
import {ORDER} from '../../../redux/order.reducer';
import BookCover from '../../general/BookCover/BookCover.component';
import Button from '../../base/Button/Button';
import Text from '../../base/Text';
import T from '../../base/T';
import Link from '../../general/Link.component';
import '../overview/dropdownList.css';
import Origin from '../../base/Origin';

const ShortListElement = props => {
  const url = `/v√¶rk/${props.element.book.pid}`;

  return (
    <div className="top-bar-dropdown-list-element">
      <div className="top-bar-dropdown-list-element--cover-image">
        <Link href={url}>
          <BookCover book={props.element.book} hideCoverText={true} />
        </Link>
      </div>
      <div className="top-bar-dropdown-list-element--text">
        <div className="top-bar-dropdown-list-element--header">
          <Link href={url}>
            <TruncateMarkup lines={1}>
              <span data-cy="shortlist-element-title">
                {props.element.book.title}
              </span>
            </TruncateMarkup>
          </Link>
        </div>
        <div className="top-bar-dropdown-list-element--taxonomy-description">
          {props.element.book.taxonomy_description}
        </div>
				<div className="top-bar-dropdown-list-element--origin">
					<Origin
						componentData={props.element.origin}
					/>
				</div>
      </div>
      {
        <i onClick={props.onClose} className="material-icons remove-btn">
          clear
        </i>
      }
    </div>
  );
};

const ShortListContent = props => {
  const emptyList = props.elements.length === 0 ? true : false;

  return (
    <div
      className={`top-bar-dropdown-list--content text-left
        ${props.expanded ? '' : ' slide-out'}
        ${emptyList ? ' empty-list' : ''}`}
    >
      <i
        onClick={props.onClose}
        className="material-icons top-bar-dropdown-list--close-btn"
      >
        clear
      </i>
      <Link href="/huskeliste">
        <Text
          type="body"
          variant="color-fersken--weight-semibold--transform-uppercase"
          className="tc"
          style={{marginBottom: '10px'}}
        >
          <T
            component="shortlist"
            name="titleWithBooks"
            vars={[`${props.elements && props.elements.length}`]}
          />
        </Text>
      </Link>
      <div
        className={
          'top-bar-dropdown-list--empty-text text-center ' +
          (emptyList ? '' : 'd-none')
        }
      >
        {emptyList && <T component="shortlist" name="emptyList" />}
      </div>
      {props.children &&
        props.children.length > 0 && (
          <div className="top-bar-dropdown-list--elements">
            <ReactCSSTransitionGroup
              transitionName="shortlist"
              transitionEnterTimeout={200}
              transitionLeaveTimeout={200}
            >
              {props.children}
            </ReactCSSTransitionGroup>
          </div>
        )}
      <div className="top-bar-dropdown-list--footer">
        <div onClick={props.onViewShortList}>
          <Button
            size="medium"
            type="tertiary"
            dataCy="shortlist-dropdown-visit-shortlist"
          >
            <T component="shortlist" name="shortlistGo" />
          </Button>
        </div>
        <div
          onClick={() => props.elements.length > 0 && props - props.orderAll()}
        >
          <Button size="medium" type="tertiary">
            <T component="shortlist" name="shortlistOrder" />
          </Button>
        </div>
      </div>
    </div>
  );
};

class ShortListDropdown extends React.Component {
  render() {
    const {expanded, elements = []} = this.props.shortListState;
    return (
      <React.Fragment>
        <div
          className={this.props.className + ' top-bar-dropdown-list'}
          onClick={() => {
            this.props.dispatch({
              type: expanded ? ON_SHORTLIST_COLLAPSE : ON_SHORTLIST_EXPAND
            });
            if (this.props.listsOverviewExpanded) {
              // collapse listOverview if expanded
              this.props.dispatch({
                type: ON_USERLISTS_COLLAPSE
              });
            }
          }}
          data-cy={this.props.dataCy}
        >
          {this.props.children}
          <span className="short-badge">{'(' + elements.length + ')'}</span>
        </div>
        <ShortListContent
          expanded={expanded}
          elements={elements}
          onClose={() => this.props.dispatch({type: ON_SHORTLIST_COLLAPSE})}
          onViewShortList={() => {
            this.props.dispatch({type: HISTORY_PUSH, path: '/huskeliste'});
            this.props.dispatch({type: ON_SHORTLIST_COLLAPSE});
          }}
          orderAll={() =>
            elements.forEach(book => {
              this.props.dispatch({type: ORDER, book: book.book});
            })
          }
        >
          {elements &&
            elements.map(element => {
              return (
                <ShortListElement
                  key={element.book.pid}
                  element={element}
                  onClose={() =>
                    this.props.dispatch({
                      type: ON_SHORTLIST_REMOVE_ELEMENT,
                      pid: element.book.pid
                    })
                  }
                />
              );
            })}
        </ShortListContent>
      </React.Fragment>
    );
  }
}
export default connect(
  // Map redux state to props
  state => {
    return {
      shortListState: state.shortListReducer,
      listsOverviewExpanded: state.listReducer.expanded
    };
  }
)(ShortListDropdown);
