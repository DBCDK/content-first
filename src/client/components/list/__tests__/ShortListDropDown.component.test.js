import React from 'react';
import {Provider} from 'react-redux';
import createStore from '../../../redux/Store';
import ShortListDropDown from '../ShortListDropDown.container';
import renderer from 'react-test-renderer';
import {ON_SHORTLIST_TOGGLE_ELEMENT, SHORTLIST_LOAD_RESPONSE} from '../../../redux/shortlist.reducer';

const createTestElement = (id) => {
  return {
    book: {
      pid: 'pid' + id,
      title: 'some title' + id,
      taxonomy_description: 'some description' + id
    },
    origin: 'Minder om noget' + id
  };
};
const addElement = (store, id) => {
  const {book, origin} = createTestElement(id);
  store.dispatch({type: ON_SHORTLIST_TOGGLE_ELEMENT, element: {book}, origin});
};

describe('ShortListDropDown', () => {
  test('short list is empty and merge modal is hidden', () => {
    const tree = renderer
      .create(<Provider store={createStore()}><ShortListDropDown /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('short list contains single element', () => {
    const store = createStore();
    addElement(store, 1);
    const tree = renderer
      .create(<Provider store={store}><ShortListDropDown /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('short list contains more than 3 elements, elements are added to top of list', () => {
    const store = createStore();
    addElement(store, 1);
    addElement(store, 2);
    addElement(store, 3);
    addElement(store, 4);
    const tree = renderer
      .create(<Provider store={store}><ShortListDropDown /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('modal is visible when merge is required', () => {
    const store = createStore();
    store.dispatch({
      type: SHORTLIST_LOAD_RESPONSE,
      localStorageElements: [createTestElement(2), createTestElement(3), createTestElement(4)],
      databaseElements: [createTestElement(1), createTestElement(2)]
    });
    const tree = renderer
      .create(<Provider store={store}><ShortListDropDown /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
