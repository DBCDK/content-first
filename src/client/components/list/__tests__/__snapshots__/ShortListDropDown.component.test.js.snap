// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortListDropDown modal is visible when merge is required 1`] = `
Array [
  <div
    className="undefined top-bar-dropdown-list"
    onClick={[Function]}
>
    <span
        className="short-badge"
    >
        (2)
    </span>
</div>,
  <div
    className="top-bar-dropdown-list--content text-left slide-out"
>
    <a
        className="list-card-nohover"
        href="/huskeliste"
        onClick={[Function]}
    >
        <h1
            className=" Heading Heading__peach-subtitle"
        >
            HUSKELISTE (
            2
            )
        </h1>
    </a>
    <div
        className="top-bar-dropdown-list--empty-text text-center d-none"
    />
    <div
        className="top-bar-dropdown-list--elements"
    >
        <span>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid1"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid1",
                                    "taxonomy_description": "some description1",
                                    "title": "some title1",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid1"
                            onClick={[Function]}
                        >
                            some title1
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description1
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget1
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid2"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid2",
                                    "taxonomy_description": "some description2",
                                    "title": "some title2",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid2"
                            onClick={[Function]}
                        >
                            some title2
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description2
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget2
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
        </span>
    </div>
    <div
        className="top-bar-dropdown-list--footer"
    >
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Gå til huskeliste
            </button>
        </div>
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Bestil hele listen
            </button>
        </div>
    </div>
</div>,
]
`;

exports[`ShortListDropDown short list contains more than 3 elements, elements are added to top of list 1`] = `
Array [
  <div
    className="undefined top-bar-dropdown-list"
    onClick={[Function]}
>
    <span
        className="short-badge"
    >
        (4)
    </span>
</div>,
  <div
    className="top-bar-dropdown-list--content text-left slide-out"
>
    <a
        className="list-card-nohover"
        href="/huskeliste"
        onClick={[Function]}
    >
        <h1
            className=" Heading Heading__peach-subtitle"
        >
            HUSKELISTE (
            4
            )
        </h1>
    </a>
    <div
        className="top-bar-dropdown-list--empty-text text-center d-none"
    />
    <div
        className="top-bar-dropdown-list--elements"
    >
        <span>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid4"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid4",
                                    "taxonomy_description": "some description4",
                                    "title": "some title4",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid4"
                            onClick={[Function]}
                        >
                            some title4
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description4
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget4
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid3"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid3",
                                    "taxonomy_description": "some description3",
                                    "title": "some title3",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid3"
                            onClick={[Function]}
                        >
                            some title3
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description3
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget3
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid2"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid2",
                                    "taxonomy_description": "some description2",
                                    "title": "some title2",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid2"
                            onClick={[Function]}
                        >
                            some title2
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description2
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget2
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid1"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid1",
                                    "taxonomy_description": "some description1",
                                    "title": "some title1",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid1"
                            onClick={[Function]}
                        >
                            some title1
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description1
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget1
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
        </span>
    </div>
    <div
        className="top-bar-dropdown-list--footer"
    >
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Gå til huskeliste
            </button>
        </div>
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Bestil hele listen
            </button>
        </div>
    </div>
</div>,
]
`;

exports[`ShortListDropDown short list contains single element 1`] = `
Array [
  <div
    className="undefined top-bar-dropdown-list"
    onClick={[Function]}
>
    <span
        className="short-badge"
    >
        (1)
    </span>
</div>,
  <div
    className="top-bar-dropdown-list--content text-left slide-out"
>
    <a
        className="list-card-nohover"
        href="/huskeliste"
        onClick={[Function]}
    >
        <h1
            className=" Heading Heading__peach-subtitle"
        >
            HUSKELISTE (
            1
            )
        </h1>
    </a>
    <div
        className="top-bar-dropdown-list--empty-text text-center d-none"
    />
    <div
        className="top-bar-dropdown-list--elements"
    >
        <span>
            <div
                className="top-bar-dropdown-list-element"
            >
                <div
                    className="top-bar-dropdown-list-element--cover-image"
                >
                    <a
                        className="list-card-nohover"
                        href="/værk/pid1"
                        onClick={[Function]}
                    >
                        <BookCover
                            book={
                                Object {
                                    "pid": "pid1",
                                    "taxonomy_description": "some description1",
                                    "title": "some title1",
                                  }
                            }
                        />
                    </a>
                </div>
                <div
                    className="top-bar-dropdown-list-element--text"
                >
                    <div
                        className="top-bar-dropdown-list-element--header"
                    >
                        <a
                            className="list-card-nohover"
                            href="/værk/pid1"
                            onClick={[Function]}
                        >
                            some title1
                        </a>
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--taxonomy-description"
                    >
                        some description1
                    </div>
                    <div
                        className="top-bar-dropdown-list-element--origin"
                    >
                        Minder om noget1
                    </div>
                </div>
                <img
                    alt="luk"
                    className="top-bar-dropdown-list-element--close-btn"
                    onClick={[Function]}
                    src="KrydsGrey.svg"
                />
            </div>
        </span>
    </div>
    <div
        className="top-bar-dropdown-list--footer"
    >
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Gå til huskeliste
            </button>
        </div>
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Bestil hele listen
            </button>
        </div>
    </div>
</div>,
]
`;

exports[`ShortListDropDown short list is empty and merge modal is hidden 1`] = `
Array [
  <div
    className="undefined top-bar-dropdown-list"
    onClick={[Function]}
>
    <span
        className="short-badge"
    >
        (0)
    </span>
</div>,
  <div
    className="top-bar-dropdown-list--content text-left slide-out"
>
    <a
        className="list-card-nohover"
        href="/huskeliste"
        onClick={[Function]}
    >
        <h1
            className=" Heading Heading__peach-subtitle"
        >
            HUSKELISTE (
            0
            )
        </h1>
    </a>
    <div
        className="top-bar-dropdown-list--empty-text text-center "
    >
        Din huskeliste er tom
    </div>
    <div
        className="top-bar-dropdown-list--footer"
    >
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Gå til huskeliste
            </button>
        </div>
        <div
            onClick={[Function]}
        >
            <button
                className=" btn Button Button__tertiary Button__medium"
            >
                Bestil hele listen
            </button>
        </div>
    </div>
</div>,
]
`;
